nodes/viz/viewer_idx28.py

```diff
@@ -157,11 +157,6 @@ def draw_buttons_ext(self, context, layout):
                 colx.scale_x = little_width
                 colx.prop(self, colprop, text="")
 
-    def update(self):
```

Comment:
compare to vd_experimental..
```python
    @property
    def fully_enabled(self):
        return "attrs" in self.inputs

    def update(self):
        if not self.fully_enabled:
            return

        try:
            socket_one_has_upstream_links = self.inputs[0].other
            socket_two_has_upstream_links = self.inputs[1].other

            if not socket_one_has_upstream_links:
                callback_disable(node_id(self))
        except:
            print('vd basic lines update holdout', self.n_id)
```

it's fully enabled when the last socket is created ( "text" in self.inputs ) and
`socket_one_has_upstream_links = self.inputs[0].other`

-------------------

not sure why this update function was so b0rked.

Reply:
you mean that could be erased? Or that i should check it?
I did not find any problem with the VD experimental. The Viewer index was disabling itself every update and it was not a good thing if you are not updating all the time

Reply:
i was being too economic with my comment. good catch btw that this update function existed in its current state. i've restated it to something more in line with what the other drawing nodes do.  (this also explains the flickering i noticed the other day.. haha )

https://github.com/nortikin/sverchok/blob/476f26dee0f3faf5ebfea52464d4b241e3a1ae0a/nodes/viz/viewer_idx28.py#L160-L172

Reply:
note to self. use fstring in self.debug, this probably throws an error at the moment if debug is on.

Reply:
I noticed UI flickring too (I‘m not sure it has the same root cause as your observed behavior) (see #3089 for details)

Reply:
no @DolphinDream , but also yes . If you push your most recent changes to your repo i'll have a look at the current code.